src/data/polynomial.lean

```diff
@@ -1835,14 +1835,6 @@ section comm_semiring
 variables [comm_semiring R] {p q : polynomial R}
 
 section aeval
-instance algebra' (R : Type u) [comm_semiring R] (A : Type v) [semiring A] [algebra R A] :
-  algebra R (polynomial A) :=
-{ smul := λ r p, algebra_map R A r • p,
-  commutes' := λ c p, ext $ λ n,
-    show (C (algebra_map R A c) * p).coeff n = (p * C (algebra_map R A c)).coeff n,
-    by rw [coeff_C_mul, coeff_mul_C, algebra.commutes],
-  smul_def' := λ c p, (C_mul' _ _).symm,
-  .. C.comp (algebra_map R A) }
```

Comment:
Could you explain to me how this instance gets provided now? I don't see it on the first pass.

Reply:
https://github.com/leanprover-community/mathlib/blob/0410946159f38b48600ed3f912eb283675ae7cbe/src/data/polynomial.lean#L581